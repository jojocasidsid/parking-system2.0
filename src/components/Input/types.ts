import React from 'react'
import { Control } from 'react-hook-form'

export interface IProps {
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	control?: Control<any, object> // To do: change any type
	defaultValue?: string
	disabled?: boolean
	error?: string
	fullWidth?: boolean
	label?: string
	name: string
	placeholder?: string
	required?: boolean
	type?: string
	startIcon?: JSX.Element | string
	endIcon?: JSX.Element | string
	multiline?: boolean
	rows?: number
	small?: boolean
	maxLength?: number
	acceptsOnly?: 'alpha' | 'alphanumeric' | 'numeric'
	hideErrorMessage?: boolean
	inputRef?: React.RefObject<HTMLInputElement | HTMLTextAreaElement | null>
}

export interface IMaskedInputProps extends IProps {
	mask: string
	maskPlaceholder?: string
}
