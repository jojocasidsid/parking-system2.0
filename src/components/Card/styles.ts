import styled from '@emotion/styled'

export const CardWrapper = styled('div')`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	padding: 20px;
	flex-direction: column;
	border-radius: 30px;
	border: 2px solid ${({ theme }) => theme.palette.secondary.main};
`
export const StyledSlotHeader = styled('div')`
	display: flex;
	justify-content: space-around;
	align-items: center;
`
