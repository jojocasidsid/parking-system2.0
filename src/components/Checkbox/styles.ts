import styled from '@emotion/styled'

export const StyledRoot = styled('div')`
	padding: 18px;
`
