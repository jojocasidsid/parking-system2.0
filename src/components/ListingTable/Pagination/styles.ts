import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export interface IPageInputProps {
	error?: string
}

export const PageInput = styled.input<IPageInputProps>`
	border-radius: 2px 0px 0px 2px;
	padding: 7px 2px 7px 0px;
	text-align: center;
	width: 100%;
	font-style: normal;
	font-size: 14px;
	line-height: 16px;
	font-weight: 500;
	font-stretch: normal;
	height: 28px;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&:active,
	&:focus {
		border: 1px solid ${({ theme }) => theme.palette.primary.main};
		outline: none;
		box-shadow: none;
	}

	&:disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	border: 1px solid
		${({ theme, error }) => {
			if (error) {
				return theme?.palette.error.main
			}
			return theme.palette.text.secondary
		}};
`

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
	font-size: 12px;
`
