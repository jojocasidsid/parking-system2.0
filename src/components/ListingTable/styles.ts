import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export const StyledTypography = styled(Typography)`
	color: #30364380;
	font-weight: bold;
	font-size: 14px;
`
