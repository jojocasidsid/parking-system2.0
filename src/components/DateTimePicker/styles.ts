import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface ITheme {}
interface IInput extends ITheme {
	errors?: boolean
	theme?: Theme
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, errors }: IInput) => {
		if (errors) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: 8px 16px 8px 16px;
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
		border-color: ${({ theme, errors }: IInput) =>
			errors
				? theme?.palette.error.main
				: theme?.palette.text.secondary} !important;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 8px;
	}

	.MuiFormHelperText-root {
		color: ${({ theme }) => theme.palette.error.main};
		margin-left: 0;
	}
`
