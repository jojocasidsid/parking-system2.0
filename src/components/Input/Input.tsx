import React from 'react'

import { Controller } from 'react-hook-form'
import { InputAdornment } from '@mui/material'
import { IProps } from './types'
import { StyledInput, StyledLabel, StyledRoot } from './styles'

const Input = ({
	control,
	defaultValue,
	error,
	label,
	name,
	disabled,
	required = false,
	placeholder,
	fullWidth,
	type,
	startIcon,
	endIcon,
	multiline,
	rows,
	maxLength,

	hideErrorMessage = false,
	inputRef,
}: IProps) => (
	<Controller
		name={name}
		control={control}
		data-testid={name}
		render={({ field: { onChange, onBlur, value } }) => (
			<StyledRoot>
				{label && (
					<StyledLabel htmlFor={name} error={Boolean(error)}>
						{label} <span>{required && '*'}</span>
					</StyledLabel>
				)}
				<StyledInput
					id={name}
					name={name}
					onChange={onChange}
					onBlur={onBlur}
					type={type}
					value={value}
					placeholder={placeholder}
					disabled={disabled}
					error={Boolean(error)}
					helperText={!hideErrorMessage ? error : undefined}
					fullWidth={fullWidth}
					inputProps={{
						maxLength,
					}}
					// eslint-disable-next-line react/jsx-no-duplicate-props
					InputProps={{
						startAdornment: startIcon && (
							<InputAdornment position='start'>{startIcon}</InputAdornment>
						),
						endAdornment: endIcon && (
							<InputAdornment position='start'>{endIcon}</InputAdornment>
						),
					}}
					multiline={multiline}
					rows={rows}
					inputRef={inputRef}
				/>
			</StyledRoot>
		)}
		defaultValue={defaultValue}
	/>
)

export default Input
