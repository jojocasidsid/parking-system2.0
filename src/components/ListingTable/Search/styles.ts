import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface IInput {
	error?: boolean
	theme?: Theme
	small?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, error }: IInput) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: ${({ small }: IInput) =>
			small ? '2px 8px 2px 8px' : '8px 12px 8px 16px'};
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme?.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px !important;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 0px;
	}

	.MuiInputBase-adornedEnd div.MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
		padding-right: 0px !important;
	}

	.MuiInputBase-adornedStart {
		padding-left: 0px !important;
	}

	.MuiInputBase-adornedStart .MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
	}

	.MuiFormHelperText-root {
		margin-left: 0;
	}
`
