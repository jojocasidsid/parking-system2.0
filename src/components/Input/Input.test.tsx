import React from 'react'

// packages
import { ThemeProvider } from '@mui/material/styles'
import { render, screen } from '@testing-library/react'
import { useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import * as yup from 'yup'
import renderer from 'react-test-renderer'

// config
import theme from 'config/theme'
import Input from './Input'

const ERROR_FIELD_REQUIRED = 'This field is required'
export const validationSchema = yup.object({
	test: yup.string().required(ERROR_FIELD_REQUIRED).nullable(),
	test1: yup.string().required(ERROR_FIELD_REQUIRED).nullable(),
})
export interface IValidationSchema
	extends yup.InferType<typeof validationSchema> {}
export const defaultValues: IValidationSchema = {
	test: '',
	test1: 'test1',
}

describe('Input component', () => {
	it('should render correctly with a name on it', () => {
		const Component = () => {
			const {
				control,

				formState: { errors },
			} = useForm({
				mode: 'onTouched',
				defaultValues,
				resolver: yupResolver(validationSchema),
			})

			return (
				<ThemeProvider theme={theme}>
					<Input
						control={control}
						error={errors.test?.message}
						fullWidth
						required
						label='Test'
						name='test'
						placeholder='test'
						type='text'
						acceptsOnly='alpha'
					/>
				</ThemeProvider>
			)
		}

		render(<Component />)

		const input = screen.getByRole('textbox') as HTMLInputElement

		expect(input).toBeVisible()
		expect(input.name).toBe('test')
	})

	it('should create snapshot correctly', () => {
		const Component = () => {
			const {
				control,

				formState: { errors },
			} = useForm({
				mode: 'onTouched',
				defaultValues,
				resolver: yupResolver(validationSchema),
			})

			return (
				<ThemeProvider theme={theme}>
					<Input
						control={control}
						error={errors.test?.message}
						fullWidth
						required
						label='Test'
						name='test'
						placeholder='test'
						type='text'
						acceptsOnly='alpha'
					/>
				</ThemeProvider>
			)
		}

		const tree = renderer.create(<Component />)

		expect(tree).toMatchSnapshot()
	})
})
