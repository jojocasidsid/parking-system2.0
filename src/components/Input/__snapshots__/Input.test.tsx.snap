// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should create snapshot correctly 1`] = `
<div
  className="css-evn1e9"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-1p8am30-MuiFormLabel-root-MuiInputLabel-root"
    htmlFor="test"
  >
    Test
     
    <span>
      *
    </span>
  </label>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-lp6u9r-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-s1ipka-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="off"
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="test"
        name="test"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="test"
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-hdw1oc"
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
